clang/test/CodeGen/ptrauth-function-init-fail.c

```diff
@@ -0,0 +1,5 @@
+// RUN: %clang_cc1 -triple arm64e-apple-ios -fptrauth-calls %s -verify -emit-llvm -o -
+
+void f(void);
+
+int *pf = (int *)&f + 1; // expected-error{{cannot compile this static initializer yet}}
```

Comment:
As commented in the test case, adding a non-zero constant to a signed pointer is unrepresentable in relocations. I agree that the diagnostic should provide more information and should inform the users that the initialization failed because of this specific limitation.

I'll see if I can improve the error message later.

Reply:
> I'll see if I can improve the error message later.

Thanks. I'm OK with having the diagnostic "as is" for this PR and submitting a patch with message improvement as another PR later.